### ISSUE TITLE
Duplexer2 fails. Duplexer passes.

### ISSUE THREAD

**Issue by aktary** [workshopper/stream-adventure#155]
On the first "Duplexer" problem, the exact same solution passes when using duplexer and fails (without a descriptive reason) when using duplexer2 (as directed in the instructions).  Below is my code:

```
var spawn = require('child_process').spawn;
var dup = require('duplexer');

module.exports = function(cmd, args){
     var child = spawn(cmd,args);
     return dup(child.stdin, child.stdout);
 }
```

**Comment by Eschon** [comment]
I have the same problem, the output of the verify script looks strange to me:

```
Administrators-Mac-mini:streams egon$ stream-adventure verify duplexer.js

TAP version 13
# (anonymous)
ok 1 stream-adventure verify YOURFILE.js
#########################################
###   YOUR SOLUTION IS NOT CORRECT!   ###
#########################################


ok 2 solution exports a function

Administrators-Mac-mini:streams egon$
```

Frome the output it looks like the 2 tests pass.

**Comment by atronov** [comment]
+1
I encounter the same issue.

**Comment by prashcr** [comment]
+1
Same issue

**Comment by prashcr** [comment]
I found it really intriguing that the solution for this exercise works perfectly, and it uses duplexer2 as well.
So I tried running both and comparing outputs.
```
➜  stream-adventure git:(master) ✗ stream-adventure run duplexer.js
gdyj
➜  stream-adventure git:(master) ✗ stream-adventure run ~/.nvm/versions/node/v4.1.1/lib/node_modules/stream-adventure/test/solutions/duplexer.js
cmjq
cbu
cvtu
czuf
cpy
cffq
cbabbs
cpuumf
cppmfbo
cffu
cbmmppo
cppq
cfrvfbui
cblf
```
The output seems to be completely different. The tests should have printed this to the console.  
  
I noticed that the official solution uses `duplexer2@0.0.2`, instead of `duplexer2@0.1.0`, which is the latest.

So I tried it for myself, and it works.
```
npm uninstall duplexer2
npm install duplexer2@0.0.2
```

**Comment by TylerMoeller** [comment]
Also hit this same issue and blocked from completing the workshop as a result. Windows 7, Node 4.2.1, npm 3.3.10.

My Solution:
```js
var spawn = require('child_process').spawn,
    duplexer = require('duplexer');

module.exports = function(cmd, args) {
  var task = spawn(cmd, args),
      stream = duplexer(task.stdin, task.stdout);
  return stream;
};
```

Output:
```bash
TAP version 13
# (anonymous)
ok 1 stream-adventure verify YOURFILE.js
ok 2 solution exports a function
not ok 3 should be equal
  ---
    operator: equal
    expected:
      'uxxi\nutm\nutdx\nunz\nunz\nukûeéx\nuhq\nublvnbm\nurstgmbgx\nurstgmbgx\nunz\nuteehhg'
    actual:
      'uxxi\nutm\nutdx'
  ...

1..3
# tests 3
# pass  2
# fail  1

#########################################
###   YOUR SOLUTION IS NOT CORRECT!   ###
#########################################
```

**Comment by klarkc** [comment]
+1 Same of @TylerMoeller

**Comment by cukejianya** [comment]
Did this ever get resolved?

**Comment by fcagalj** [comment]
+1

**Comment by cukejianya** [comment]
I ran the solution myself and it also failed. 

```
stream-adventure verify ~/.npm/lib/node_modules/stream-adventure/test/solutions/duplexer.js

TAP version 13
# (anonymous)
ok 1 stream-adventure verify YOURFILE.js
ok 2 solution exports a function
not ok 3 should be equal
  ---
    operator: equal
    expected:
      'gwûqéj\ngnyj\ngnxhzny\ngfeffw\ngfpj\ngjjyqj\ngjjy\ngfqqtts\ngjju\ngjvzjfym\ngdyj\ngfeffw'
    actual:
      'gwûqéj\ngnyj\ngnxhzny'
    at: ConcatStream.<anonymous> (/Users/Chinedum/.npm/lib/node_modules/stream-adventure/node_modules/concat-stream/index.js:36:43)
  ...

1..3
# tests 3
# pass  2
# fail  1

#########################################
###   YOUR SOLUTION IS NOT CORRECT!   ###
#########################################
```

**Comment by deubaka** [comment]
Got it working now with multiple options. 

The first one is to downgrade `stream-adventure` to `v3.0.0`, everything works should work there. 

The second be overriding `stream-combiner2` to use `v1.0.2` only as follow,
- Removed module references to `stream-combiner2` latest (v1.1.1), that means removing it in your libs both global and local
- Within the `stream-adventure` module directory, re-install `stream-combiner2` at `v1.0.2` via `npm install  --save stream-combiner@1.0.2`
- Re-run verify and should show a solution passed now

**Comment by cukejianya** [comment]
@deubaka Thank you!. The second option worked!!

**Comment by mofodox** [comment]
@deubaka @cukejianya Sweet! +1 to option 2 👍

**Comment by brookinc** [comment]
@deubaka Also +1 to option 1. :+1: :smile:

**Comment by benwiley4000** [comment]
I just ran `npm uninstall stream-combiner2` in the `stream-adventure` module directory, and `npm install -g stream-combiner2@1.0.2`. Thanks @deubaka.

A little confused about why this issue was occurring, though. I assume there must be a *proper* way to reconfigure our code to work with the latest releases/APIs? I'd rather learn what's up-to-date.

**Comment by beedaan** [comment]
@benwiley4000, I tried to install stream-combiner2 at version 1.0.2 and I was unsuccessful
```
$ npm install --save stream-combiner@1.0.2
npm ERR! Error: version not found: stream-combiner@1.0.2
```
Any advice?

**Comment by beedaan** [comment]
@benwiley4000 Thank you!  That worked for me! :-)

**Comment by deubaka** [comment]
@beedaan I believe there is already a pull request from others not sure though. However, I agree with @benwiley4000 that there needs to be a proper solution other that having an outdated dependency as a temporary workaround. :D

**Comment by ralphtheninja** [comment]
This should be fixed now. Can someone confirm this?

**Comment by beedaan** [comment]
@ralphtheninja I just verified that the issue is fixed.

**Comment by jandrew** [comment]
This issue happened to me as well (including the defined solution failing) on windows 10.  The fix for me was to rewrite the return duplexer line as;

    return duplexer({objectMode:false}, ps.stdin, ps.stdout);

**Comment by ccarruitero** [comment]
fixed by #172
